homeassistant/components/vodafone_station/device_tracker.py

```diff
@@ -0,0 +1,127 @@
+        )
+    )
+
+    async_update_router()
+
+
+@callback
+def async_add_new_tracked_entities(
+    coordinator: VodafoneStationRouter,
+    async_add_entities: AddEntitiesCallback,
+    tracked: set[str],
+) -> None:
+    """Add new tracker entities from the router."""
+    new_tracked = []
+
+    _LOGGER.debug("Adding device trackers entities")
+    for mac, device_info in coordinator.data.devices.items():
+        if mac in tracked:
+            continue
+        _LOGGER.debug("New device tracker: %s", device_info.device.name)
+        new_tracked.append(VodafoneStationTracker(coordinator, device_info))
+        tracked.add(mac)
+
+    async_add_entities(new_tracked)
+
+
+class VodafoneStationTracker(CoordinatorEntity[VodafoneStationRouter], ScannerEntity):
+    """Representation of a Vodafone Station device."""
+
+    def __init__(
+        self, coordinator: VodafoneStationRouter, device_info: VodafoneStationDeviceInfo
+    ) -> None:
+        """Initialize a Vodafone Station device."""
+        super().__init__(coordinator)
+        self._coordinator = coordinator
+        device = device_info.device
+        mac = device.mac
+        self._device_mac = mac
+        self._attr_unique_id = mac
+        self._attr_name = device.name or mac.replace(":", "_")
+
+    @property
+    def _device_info(self) -> VodafoneStationDeviceInfo:
+        """Return fresh data for the device."""
+        return self.coordinator.data.devices[self._device_mac]
+
+    @property
+    def _device(self) -> VodafoneStationDevice:
+        """Return fresh data for the device."""
+        return self.coordinator.data.devices[self._device_mac].device
+
+    @property
+    def is_connected(self) -> bool:
+        """Return true if the device is connected to the network."""
+        return self._device_info.home
+
+    @property
+    def source_type(self) -> SourceType:
+        """Return the source type."""
+        return SourceType.ROUTER
+
+    @property
+    def hostname(self) -> str | None:
+        """Return the hostname of device."""
+        return self._attr_name
+
+    @property
+    def icon(self) -> str:
+        """Return device icon."""
+        return "mdi:lan-connect" if self._device.connected else "mdi:lan-disconnect"
+
+    @property
+    def ip_address(self) -> str | None:
+        """Return the primary ip address of the device."""
+        return self._device.ip_address
+
+    @property
+    def mac_address(self) -> str:
+        """Return the mac address of the device."""
+        return self._device_mac
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return additional attributes of the device."""
+        connection_type = self._device.connection_type
+        attrs: dict[str, Any] = {"connection_type": connection_type}
+        if "Wifi" in connection_type:
+            attrs["wifi_band"] = self._device.wifi
+        attrs["last_time_reachable"] = self._device_info.update_time
```

Comment:
Let's drop this from the initial PR as it should be an optional sensor they can turn off to reduce state writes

Reply:
Done